### DISCUSSION TITLE
Colab TPUv2 session crash on diffusers import

### DISCUSSION THREAD

**Discussion by dan-crdll** [huggingface/diffusers#8314] [Q&A] [answered]
To reproduce the problem:

```
!pip install transformers
!pip install diffusers
!pip install tensorboard
!pip install --upgrade torch torchvision torchaudio
!pip install git+https://github.com/openai/CLIP.git@a1d071733d7111c9c014f024669f959182114e33

import jax.tools.colab_tpu

jax.tools.colab_tpu.setup_tpu()

import torch
import torchvision
import torch_xla
import torch_xla.core.xla_model as xm

from diffusers import UNet2DConditionModel, AutoencoderKL, PNDMScheduler
import clip

device = xm.xla_device()
print(f'TPU Device: {device}')
```

Session crash with unknown problem when importing diffusers

**Comment by tolgacangoz** [comment]
It seems that classes of `diffusers` and `torch_xla` cannot be imported in the same Colab TPUv2 session currently :thinking:.
Cc: @pcuenca

**Reply by pcuenca** [reply]
Interesting. I could replicate a similar crash by importing diffusers first (it works) and then importing `torch_xla` (it crashed). This sounds like an issue for Colab, wouldn't you think?

```
!pip install diffusers

import jax.tools.colab_tpu
jax.tools.colab_tpu.setup_tpu()

from diffusers import UNet2DConditionModel, AutoencoderKL, PNDMScheduler
import torch_xla
```

(The same crash happens even if you don't call `jax.tools.colab_tpu.setup_tpu()`)

**Comment by suzukimain** [comment, marked as answer]
I had the same problem, but doing this solved it.
https://github.com/huggingface/transformers/issues/30516#issuecomment-2100689652

**Reply by dan-crdll** [reply]
Thanks, this solved it!
